### ISSUE TITLE
pslegend crashes with a segmentation fault

### ISSUE THREAD

**Issue by claudiodsf** [GenericMappingTools/gmt#1096]
**Description of the problem**

`pslegend` crashes with a segmentation fault. 

**Full script that generated the error**

```
gmt pslegend -JX10 -R0/1/0/1 -D0/0/1 << EOF > plot.ps
T blabla
G 0.5
T toto
EOF
```

Removing `G 0.5`, the script works.

**Full error message**

```
ERROR: Caught signal number 11 (Segmentation fault) at
0   libsystem_platform.dylib            0x00007fff63f918c0 _platform_strncpy + 64
1   ???                                 0x0000000000000000 0x0 + 0
Stack backtrace:
0   libgmt.6.dylib                      0x000000010b6dfe82 sig_handler + 558
1   libsystem_platform.dylib            0x00007fff63f93b5d _sigtramp + 29
2   ???                                 0x0000000000000000 0x0 + 0
3   libgmt.6.dylib                      0x000000010b6f3054 GMT_Get_Record + 91
4   libgmt.6.dylib                      0x000000010b94f03a GMT_pstext + 8006
5   libgmt.6.dylib                      0x000000010b6f7c29 GMT_Call_Module + 626
6   libgmt.6.dylib                      0x000000010b96c6c3 GMT_pslegend + 31566
7   libgmt.6.dylib                      0x000000010b6f7c29 GMT_Call_Module + 626
8   gmt                                 0x000000010b6d5097 main + 1069
9   libdyld.dylib                       0x00007fff63da83d5 start + 1
10  ???                                 0x0000000000000005 0x0 + 5
```

**System information**

* Operating system: macOS 10.14.5
* Version of GMT: 6.0.0rc2 (same problem with rc1, works with GMT5.4.5)

**Comment by PaulWessel** [comment]
Thanks for the report, I found and fixed the problem via #1097 so once approved and merged it will be in the 6.0 branch and master.

**Comment by claudiodsf** [comment]
Thanks Paul ! I confirm that #1097 fixes the issue.
